src/Desktop/UIAutomation/CustomObjects/Registrar.cs

```diff
@@ -9,21 +14,59 @@ namespace Axe.Windows.Desktop.UIAutomation.CustomObjects
 {
     public class Registrar : IDisposable
     {
-        CUIAutomationRegistrar _uiaRegistrar = new CUIAutomationRegistrar();
+        private IUIAutomationRegistrar _uiaRegistrar;
+        private Action<int, ITypeConverter> _callback;
         private bool disposedValue;
 
-        public int RegisterCustomProperty(Guid id, string name, UIAutomationType type)
+        public Registrar()
+            : this(new CUIAutomationRegistrar(), new Action<int, ITypeConverter>(A11yProperty.RegisterCustomProperty)) { }
+
+        public void RegisterCustomProperty(CustomProperty prop)
         {
+            if (prop == null) throw new ArgumentNullException(nameof(prop));
             UIAutomationPropertyInfo info = new UIAutomationPropertyInfo
             {
-                guid = id,
-                pProgrammaticName = name,
-                type = type,
+                guid = prop.Guid,
+                pProgrammaticName = prop.ProgrammaticName,
+                type = GetUnderlyingUIAType(prop.Type)
             };
 
             _uiaRegistrar.RegisterProperty(ref info, out int dynamicId);
+            _callback(dynamicId, CreateTypeConverter(prop.Type));
+        }
 
-            return dynamicId;
+        internal Registrar(IUIAutomationRegistrar uiaRegistrar, Action<int, ITypeConverter> callback)
```

Comment:
Would it make sense to move the internal ctor to appear immediately after the public ctor? That's how we usually manage ctors when we support dependency injection this way. I'm guessing that you followed a pattern of putting public methods first. We don't really have a policy about sorting by visibility, especially since it can change at any time